src/create_disk_se.sh

```diff
@@ -0,0 +1,410 @@
+#!/bin/sh
+# This script is run in supermin to create a Fedora CoreOS style
+# disk image, very much in the spirit of the original
+# Container Linux (orig CoreOS) disk layout, although adapted
+# for OSTree, and using XFS for /, among other things.
+# Some more background in https://github.com/coreos/fedora-coreos-tracker/issues/18
+# The layout is intentionally not very configurable at this time,
+# although see also https://github.com/coreos/coreos-assembler/pull/298
+# For people building "derived"/custom FCOS-like systems, feel free to file
+# an issue and we can discuss configuration needs.
+set -euo pipefail
+
+# This fixed UUID is detected in ignition-dracut and changed
+# on firstboot:
+# https://github.com/coreos/ignition-dracut/blob/6136be3d9d38d7926a61cd4d1b4ba5f9baf0892f/dracut/30ignition/coreos-gpt-setup.sh#L7
+uninitialized_gpt_uuid="00000000-0000-4000-a000-000000000001"
+# These UUIDs should be changed by code in fedora-coreos-config on firstboot, see
+# https://github.com/coreos/fedora-coreos-tracker/issues/465
+bootfs_uuid="96d15588-3596-4b3c-adca-a2ff7279ea63"
+rootfs_uuid="910678ff-f77e-4a7d-8d53-86f2ac47a823"
+deploy_root=
+
+usage() {
+    cat <<EOC
+${0} creates a supermin virtual machine to create a
+Fedora CoreOS style disk image from an OSTree.
+
+Options:
+    --config: JSON-formatted image.yaml
+    --disk: disk device to use
+    --help: show this help
+    --kargs: kernel CLI args
+
+You probably don't want to run this script by hand. This script is
+run as part of 'coreos-assembler build'.
+EOC
+}
+
+#SecureExecution variables
+secure_execution_hostkey="/dev/disk/by-id/virtio-hostkey"
+secure_execution_initrd="/tmp/initrd.secure"
+secure_execution_bootfs_key="boot"
+secure_execution_bootfs_label="crypt_bootfs"
+secure_execution_bootfs_device="/dev/mapper/crypt_bootfs"
+
+secure_execution_rootfs_key="root"
+secure_execution_rootfs_label="crypt_rootfs"
+secure_execution_rootfs_device="/dev/mapper/crypt_rootfs"
+
+regenerate_initramfs() {
+    old_initrd=$1
+    new_initrd=$2
+    # Unpacking existing initramdisk
+    currdir=$PWD
+    workdir=$(mktemp -d -p /tmp se-initramfs-XXXXXX)
+    cd ${workdir}
+    gzip -cd ${old_initrd} | cpio -imd --quiet
+    # Adding LUKS root key and crypttab config
+    mkdir -p etc/luks
+    cp -f $deploy_root/etc/luks/* etc/luks/
+    cp -f $deploy_root/etc/crypttab etc/
+    # Creating new initramdisk image
+    find . | cpio --quiet -H newc -o | gzip -9 -n >> ${new_initrd}
```

Comment:
This should probably be `find . -mindepth 1`. See https://github.com/coreos/coreos-assembler/pull/1487.